Text/CustomEntitySearch/CustomEntitySearch.cs

```diff
@@ -95,5 +128,310 @@ public static class CustomEntitySearch
+                            wordFound.GetStringBuilder().Clear();
+                            currMismatch = 0;
+                            continue;
+                        }
+
+                        if (currMismatch <= leniency)
+                        {
+                            if (wordCharArray[currWordCharIndex] == textCharArray[currTextCharIndex])
+                            {
+                                if (textCharArray[currTextCharIndex].IsDelineating() &&
+                                    (bestMismatchPreWhitespace == -1 || bestMismatchPreWhitespace > currMismatch + (wordCharArray.Length - currWordCharIndex)))
+                                {
+                                    whitespaceOffset = currTextCharIndex;
+                                    bestMismatchPreWhitespace = currMismatch + (wordCharArray.Length - currWordCharIndex);
+                                }
+                                wordFound.Write(text[currTextCharIndex]);
+                                currTextCharIndex++;
+                                currWordCharIndex++;
+                            }
+                            else
+                            {
+                                double potTextMismatch = (textCharArray[currTextCharIndex].IsAccent()) ? 0.5 : 0;
+                                double potWordMismatch = (wordCharArray[currWordCharIndex].IsAccent()) ? 0.5 : 0;
+                                // fuzzy situation?
+                                // accent case adds 0.5
+                                if (String.Compare(wordCharArray[currWordCharIndex].ToString(), textCharArray[currTextCharIndex].ToString(),
+                                    CultureInfo.CurrentCulture, CompareOptions.IgnoreNonSpace) == 0)
+                                {
+                                    currMismatch += 0.5;
+                                    wordFound.Write(text[currTextCharIndex]);
+                                    currTextCharIndex++;
+                                    currWordCharIndex++;
+                                }
+                                else if (potWordMismatch > 0 && potTextMismatch == 0)
+                                {
+                                    currMismatch += 0.5;
+                                    currWordCharIndex++;
+                                }
+                                else if (potWordMismatch == 0 && potTextMismatch > 0)
+                                {
+                                    currMismatch += 0.5;
+                                    wordFound.Write(text[currTextCharIndex]);
+                                    currTextCharIndex++;
+                                }
+                                else
+                                {
+                                    int offsetWord = 0;
+                                    bool trueWordComp = false;
+                                    bool trueTextComp = false;
```

Comment:
these variable names are kind of confusing, these look like "foundMatchInWordLookAhead" and "foundMatchInTextLookAhead"?